src/index.tsx

```diff
@@ -106,26 +115,37 @@ class LoadingBar extends Component<IProps, IState> {
         clearInterval(interval);
       }
     }, 1000);
+    this.setState({ progress: random, interval });
   };
 
   public staticStart = (startingValue: number) => {
+    if (this.state.interval) {
+      clearInterval(this.state.interval);
+    }
     const random = startingValue || this.randomInt(30, 50);
 
-    this.setState({ progress: random }, () => {
+    this.setState({ progress: random, interval: null }, () => {
       this.onProgressChange();
     });
   };
 
   public complete = () => {
-    this.setState({ progress: 100 }, () => {
+    if (this.state.interval) {
+      clearInterval(this.state.interval);
+    }
+    this.setState({ progress: 100, interval: null }, () => {
       this.onProgressChange();
     });
   };
 
   private onLoaderFinished = () => {
     if (this.props.onLoaderFinished) this.props.onLoaderFinished();
 
-    this.setState({ progress: 0 }, () => {
+    if (this.state.interval) {
```

Comment:
I think we don't need this here. Since continues start is used with refs. And this method is just a event.